### ISSUE TITLE
Unused private properties when using magic __get

### ISSUE THREAD

**Issue by mrtnzlml** [slevomat/coding-standard#56]
Hi, I want to use magic method for simple properties in object like in the example bellow. You can access them using `{$order->orderNumber}` in template thanks to the `@method` annotation. With this approach it's super easy to write custom getter without breaking code in the template. It's somehow possible to take `@method` annotations into account? Or should I think about better approach? Thanks! :)

```php
<?php

/**
 * @method getOrderNumber()
 */
class OrderDTO extends \Nette\Object
{

	private $orderNumber; //Warning: unused private element

	public function __construct(\App\Orders\Order $order)
	{
		$this->orderNumber = $order->getId();
	}

}
```

**Comment by ondrejmirtes** [comment]
Hi,
`UnusedPrivateElementsSniff` supports configuring "always used" properties with listing annotations above such properties that mark them as used, so you can do for example this:

```
/**
 * @get
 */
private $orderNumber;
```

Also, suffixes (like `Timestamp` in this example) that such properties have in common can be used:

```
private $createdTimestamp;
```

Version 2.0 currently [in development](https://github.com/slevomat/coding-standard/pull/53) also supports annotations that can suppress checks on certain properties:

```
/**
 * @phpcsSuppress SlevomatCodingStandard.Classes.UnusedPrivateElements.writeOnlyProperty
 */
private $writeOnlyPropertyWithSuppress;
```

So the sniff does not offer anything that would work out of the box with your syntax, but there are several options you can go with.

But I'd recommend against these kind of practices (widely used magic behaviour with `__call`) in your code because the application has to figure out a lot of stuff in runtime for them to work. I'd guess that the logic in `Nette\Object` is called thousands of times and slowing your application a lot in the end.

If you want to save yourself some typing, you can generate those getters in your IDE, or generate classes to be used in your application from those "source" classes during compile time, i. e. something like Doctrine proxies work.

**Comment by mrtnzlml** [comment]
Thanks, I'm using `alwaysUsedPropertiesAnnotations` for `@ORM\Column` already. This approach is super easy for frontend developers (`{$order->identifier}` means `OrderDTO::getIdentifier()` and `$identifier` is still private) and I wanted to use something like `LazyAccessor` therefore performance is not an issue. But the proxy pattern seems like an interesting idea. For now I'm going to use a lot of getters (even though it seems mostly unnecessary for DTO objects)...

Thank you @ondrejmirtes.
